src/Core/Managed/Shared/Channel/TelemetryBuffer.cs

```diff
@@ -15,19 +15,21 @@ internal class TelemetryBuffer
         public Action OnFull;
 
         private const int DefaultCapacity = 500;
+        private const int DefaultMaximumUnsentBacklogSize = 1000000;        
```

Comment:
For the in memory channel this should be 1001.
For the server telemetry channel it should be 1000000.